packages/rancher-windows-gmsa/charts/templates/issuer.yaml

```diff
@@ -0,0 +1,50 @@
+{{- if .Values.certificates.certManager.enabled -}}
+  {{- $certmanagerVer :=  split "." .Values.certificates.certManager.version -}}
+  {{- if or (.Capabilities.APIVersions.Has "cert-manager.io/v1") (and (gt (len $certmanagerVer._0) 0) (eq (int $certmanagerVer._0) 1) (ge (int $certmanagerVer._1) 0)) }}
+apiVersion: cert-manager.io/v1
+  {{- else if or (.Capabilities.APIVersions.Has "cert-manager.io/v1beta1") (and (gt (len $certmanagerVer._0) 0) (eq (int $certmanagerVer._0) 0) (ge (int $certmanagerVer._1) 16)) }}
+apiVersion: cert-manager.io/v1beta1
+  {{- else if or (.Capabilities.APIVersions.Has "cert-manager.io/v1alpha2") (and (gt (len $certmanagerVer._0) 0) (eq (int $certmanagerVer._0) 0) (ge (int $certmanagerVer._1) 11)) }}
+apiVersion: cert-manager.io/v1alpha2
+  {{- else if or (.Capabilities.APIVersions.Has "certmanager.k8s.io/v1alpha1") (and (gt (len $certmanagerVer._0) 0) (eq (int $certmanagerVer._0) 0) (lt (int $certmanagerVer._1) 11)) }}
+apiVersion: certmanager.k8s.io/v1alpha1

```

Comment:
This should definitely be in a helper function since it is called twice and has complex logic

IMO you should be able to call it like:

```yaml
apiVersion: {{ include "cert-manager.apiVersion" }}
```

Reply:
This logic also seems way to complicated; why do we need to support multiple cert-manager versions? Is this specifically requested or added to future-proof?

I think it's sufficient enough that we should just allow the user to pass in the apiVersion as a string to override if they want, otherwise, we just always do `cert-manager.io/v1`

Reply:
@aiyengar2 I pulled this logic straight from the externalip-webhook chart we already have.

Reply:
Hmm, in that case using the logic is fine but it should definitely be in a helper function, not repeated twice in the code.